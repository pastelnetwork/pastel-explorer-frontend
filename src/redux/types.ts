import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { Action } from 'redux';
import { RootState } from '@redux/store';

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;

export type AppThunkDispatch = ThunkDispatch<RootState, unknown, Action<string>>;
