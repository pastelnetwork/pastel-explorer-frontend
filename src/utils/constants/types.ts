import { LabelKeyObject } from 'react-csv/components/CommonPropTypes';
import { PeriodTypes, TGranularity } from '@utils/helpers/statisticsLib';
import { TStatisticsInfo } from '@utils/types/IStatistics';

export type TThemeColor = {
  name: string;
  backgroundColor: string;
  lineColor?: string;
  splitLineColor: string;
  color: string;
  stack?: string;
  smooth?: boolean;
};

export type TThemeInitOption = {
  theme?: TThemeColor | null;
  data?: number[][];
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  chartName: string;
  minY: number;
  maxY: number;
};

export type TLineChartProps = {
  chartName: string;
  period?: PeriodTypes;
  dataX?: string[];
  dataY?: number[];
  dataY1?: number[];
  dataY2?: number[];
  title?: string;
  granularity?: TGranularity;
  info: TStatisticsInfo;
  granularities?: TGranularity[];
  offset: number;
  periods?: PeriodTypes[];
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
};

export type TCsvHeaderType = {
  [key: string]: LabelKeyObject[];
};

export type TScatterChartProps = {
  chartName: string;
  data: number[][];
  dataX: string[];
  title?: string;
  period?: PeriodTypes;
  info: TStatisticsInfo;
  offset: number;
  periods: PeriodTypes[];
  handleBgColorChange: (_color: string) => void;
  handlePeriodFilterChange?: (_period: PeriodTypes) => void;
  handleGranularityFilterChange?: (_granularity: TGranularity) => void;
};
