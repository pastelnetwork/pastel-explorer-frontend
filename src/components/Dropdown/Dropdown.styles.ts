import styled from 'styled-components';

export const Wrapper = styled.div`
  display: inline-flex;
  align-items: center;

  .MuiSelect-icon {
    top: 46%;
    right: 4px;
    transform: translateY(-50%);
  }

  &.hidden-fieldset {
    fieldset {
      display: none;
    }
  }

  .MuiSelect-select {
    width: 150px;
    padding: 6px 28px 7px 10px !important;
    border: 1px solid ${props => props.theme.dropdown.border};
    color: ${props => props.theme.dropdown.color};
    border-radius: 4px;

    &:focus {
      background: transparent;
    }
  }

  .MuiInput-underline {
    &:after,
    &:before {
      display: none;
    }

    &:hover {
      &:not(.Mui-disabled) {
        &:before {
          border-bottom-width: 1px;
        }
      }
    }
  }

  svg {
    color: ${props => props.theme.dropdown.color};
  }
`;

export const Label = styled.div`
  margin-right: 10px;
  color: ${props => props.theme.dropdown.color};
  font-weight: 500;
  white-space: nowrap;
`;
