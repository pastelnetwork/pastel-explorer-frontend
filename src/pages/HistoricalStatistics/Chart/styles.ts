import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
