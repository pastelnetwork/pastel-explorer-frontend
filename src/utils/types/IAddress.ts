export interface IAddressData {
  amount: number;
  timestamp: number;
  transactionHash: string;
  direction: string;
}

export interface IAddress {
  address: string;
  incomingSum: number;
  outgoingSum: number;
  data: Array<IAddressData>;
}
