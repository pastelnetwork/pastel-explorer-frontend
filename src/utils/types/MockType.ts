import { TFunction } from 'i18next';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export type MyMockType = jest.Mock<any, any, any> &
  TFunction<['translation', ...string[]], undefined>;
