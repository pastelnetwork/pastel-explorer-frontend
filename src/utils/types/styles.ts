/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
import { SpacingProps } from '@material-ui/system';
import { Theme } from '@material-ui/core';

export type GlobalStyleProps = {
  theme: Theme & { palette: any };
};

export interface MuiButtonSpacingType extends SpacingProps {
  component?: React.PropsWithoutRef<{}>;
  to?: string;
}
