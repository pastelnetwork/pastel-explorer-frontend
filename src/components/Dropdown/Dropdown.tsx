import * as React from 'react';

import MenuItem from '@mui/material/MenuItem';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import Checkbox from '@mui/material/Checkbox';
import ListItemText from '@mui/material/ListItemText';
import Input from '@mui/material/Input';

import { translateDropdown } from '@utils/helpers/i18n';

import * as Styles from './Dropdown.styles';

export type OptionsProps = {
  value: string | number;
  name: string;
};

type DropdownProps = {
  value: string;
  onChange: (_value: SelectChangeEvent) => void;
  options: OptionsProps[];
  label?: React.ReactNode;
  classNameWrapper?: string;
};

export const Dropdown = ({
  value,
  onChange,
  options,
  label,
  classNameWrapper,
}: DropdownProps): JSX.Element => {
  return (
    <Styles.Wrapper className={classNameWrapper}>
      {label ? <Styles.Label>{label}</Styles.Label> : null}
      <Select
        value={value as string}
        onChange={onChange}
        MenuProps={{
          style: {
            maxHeight: '50vh',
          },
        }}
      >
        {options.map((option: OptionsProps) => (
          <MenuItem value={option.value} key={option.value}>
            {option.name}
          </MenuItem>
        ))}
      </Select>
    </Styles.Wrapper>
  );
};

Dropdown.defaultProps = {
  label: undefined,
  classNameWrapper: '',
};

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 250,
    },
  },
};

interface ITagDropdown {
  defaultValues?: string[];
  onChange?: (_values: string[]) => void;
  placeholder?: string;
  options: OptionsProps[];
}

export const TagDropdown: React.FC<ITagDropdown> = ({
  defaultValues = [],
  onChange,
  placeholder,
  options,
}) => {
  const [values, setValues] = React.useState<string[]>(defaultValues);

  const handleChange = (event: SelectChangeEvent<typeof values>) => {
    setValues(
      typeof event.target.value === 'string'
        ? event.target.value.split(',')
        : (event.target.value as string[]),
    );
  };

  const handleClose = () => {
    if (onChange) {
      onChange(values);
    }
  };

  return (
    <Styles.Wrapper>
      <Select
        multiple
        displayEmpty
        value={values}
        onChange={handleChange}
        onClose={handleClose}
        input={<Input />}
        renderValue={selected => {
          if (!(selected as string[]).length) {
            return placeholder;
          }
          const items = options
            .filter(o => o.value && (selected as string[]).indexOf(o.value.toString()) !== -1)
            .map(o => translateDropdown(o.name));
          return items.join(', ');
        }}
        MenuProps={MenuProps}
      >
        {options.map(option => (
          <MenuItem key={option.value} value={option.value}>
            <Checkbox checked={values.indexOf(option.value.toString()) > -1} />
            <ListItemText primary={translateDropdown(option.name)} />
          </MenuItem>
        ))}
      </Select>
    </Styles.Wrapper>
  );
};

TagDropdown.defaultProps = {
  defaultValues: [],
  placeholder: '',
  onChange: () => {},
};
