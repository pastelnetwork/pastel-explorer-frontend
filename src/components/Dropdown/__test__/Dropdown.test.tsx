import { shallow } from 'enzyme';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import 'jest-styled-components';

import { MyMockType } from '@utils/types/MockType';
import i18next from '../../../utils/helpers/i18n';
import { Dropdown } from '../Dropdown';
import * as Styles from '../Dropdown.styles';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
}) as MyMockType;

describe('components/Dropdown', () => {
  const value = '1';
  const options = [
    {
      name: 'Option 1',
      value: '1',
    },
    {
      name: 'Option 2',
      value: '2',
    },
  ];
  const onChange = () => {
    // TODO
  };
  const wrapper = shallow(<Dropdown value={value} options={options} onChange={onChange} />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <DateRangeIcon>', () => {
    expect(wrapper.find(MenuItem).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Select>', () => {
    expect(wrapper.find(Select).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.Wrapper>', () => {
    expect(wrapper.find(Styles.Wrapper).length).toBeGreaterThanOrEqual(1);
  });
});
