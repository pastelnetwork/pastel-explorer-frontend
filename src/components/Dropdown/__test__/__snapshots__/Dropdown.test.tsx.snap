// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropdown renders correctly 1`] = `ShallowWrapper {}`;
